localstack/utils/aws/aws_stack.py

```diff
@@ -244,8 +244,8 @@ def connect_to_service(service_name, client=True, env=None, region_name=None, en
                 endpoint_url = backend_url
         config = config or botocore.client.Config()
         # configure S3 path style addressing
-        if service_name == 's3':
-            config.s3 = {'addressing_style': 'path'}
+        # if service_name == 's3':
+        #     config.s3 = {'addressing_style': 'path'}
```

Comment:
I'm actually a bit surprised that this is already working. Wouldn't we also need to change `get_local_service_url(..)` to return a URL like `http://localhost.localstack.cloud:4566`? I'm just wondering if boto3 is actually using path style addressing as a default/fallback, in case the endpoint URL is `http://localhost:4566`..? 

Otherwise, with the current configuration in this PR, I'd assume boto3 should make a request to a host name like `http://mybucket.s3.us-east-1.localhost:4566`, which should fail as this host cannot be resolved.